### PR TITLE
Add many to many example with robots and their parts

### PR DESCRIPTION
**Pull request by evolic**
Added working example of saving Many-To-Many relationship when
* creating new entity
  * all robot parts are inserted as referring to existing or new parts
* updating existing entity
  * we need to figure our which parts are
    * the same
    * removed, so we delete then
    * added, so they will be inserted as referring to existing or new parts

### CODE REVIEW

File: app/controllers/RobotsController.php [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+                }
+            }
+        }
+
+        $this->view->robot = $robot;
+
+        $this->view->form = new RobotsForm($robot, $options);
+    }
+
+    protected function processRobotParts($robot, $oldPartNames, $newPartNames)
```
Comment (evolic) [inline app/controllers/RobotsController.php]: Method finds 3 states of the relationship:
* not changed,
* to delete
* to add

**Comment by sergeysviridenko** [comment]
@evolic Thank you for this PR. I'll try to take a look asap.

**Comment by sergeyklay** [comment]
@evolic We don't add changes directly to stable branches or `master`. Could you please send this PR to the `development` branch?

**Comment by ruudboon** [comment]
Closing this expired pull

**Pull request closed without merging**